### PR TITLE
Moved All Evaluation Functions in STP To The AI Level

### LINKED ISSUE #795: Move All Evaluation Functions in STP To The AI Level
### Description of the task
We currently have two evaluation folders: `ai/evaluation` and `ai/hl/stp/evaluation`. We should move all evaluation functions to the `ai/evaluation` folder, as there are no "STP-specific" ones.

### Acceptance criteria

- [ ] Move all evaluation functions in `ai/hl/stp/evaluation` to `ai/evaluation`

### PR DESCRIPTION
**Pull request by amrm3lm**
### Description
Moved all evaluation files from `ai/hl/stp/evaluation` to `ai/evaluation` . 
Added the contents of `ai/hl/stp/evaluation/BUILD` to `ai/evaluation/BUILD`. 
Changed references to `ai/hl/stp/evaluation` functions found in `ai/evaluation/BUILD`, `ai/hl/stp/tactic/BUILD`, and `ai/hl/stp/play/BUILD`. 

### Testing Done
ran `bazel build //...` and build was successful. 
ran `bazel test //...` and all tests passed. 

### Resolved Issues
resolves #795 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by henryymliu** [comment]
First pass LGTM, but I would like #867 to go through first before this PR; hopefully you won't have to deal with merge conflicts

**Comment by garethellis0** [comment]
Gonna give @akhilveeraghanta the first pass for this review.

**Comment by henryymliu** [comment]
@amrm3lm can you pull from upstream and resolve conflicts?

**Comment by MathewMacDougall** [comment]
Nice work fixing conflicts. @akhilveeraghanta @henryymliu please review.

File: src/software/ai/evaluation/possession.cpp
Comment (henryymliu) [inline src/software/ai/evaluation/possession.cpp]: Duplicate includes here.

**Review by henryymliu — CHANGES_REQUESTED** [review]
Just one change, everything else LGTM

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada: aside from Henry's comment

**Comment by garethellis0** [comment]
Oh boy, looks like one file has a merge conflict. :grimacing:

**Comment by MathewMacDougall** [comment]
Ah looks like more evaluation functions changed (I think from all the RoboCup related changes). @amrm3lm Can you resolve the merge conflicts again :grimacing: ?

**Review by akhilveeraghanta — DISMISSED** [review]
Looks Good :tm: after merge conflicts fixed

File: src/software/ai/evaluation/possession.cpp
Comment (amrm3lm) [reply src/software/ai/evaluation/possession.cpp]: nice spot, I removed them

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1: Thanks for persevering through the conflicts

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1: Thanks for persevering through the conflicts

**Comment by MathewMacDougall** [comment]
@henryymliu will need your re-approval

**Review by akhilveeraghanta — APPROVED** [review]
Looks Good :tm:

**Review by henryymliu — APPROVED** [review]
🚢🤡

**Pull request merged**
